### ISSUE TITLE
Report table never accessed when it only got index assignments

### ISSUE THREAD

**Issue by bjorn** [mpeterv/luacheck#61]
The following snippet does not do anything useful with `foo`, yet it is not marked as "unused":
```lua
local foo = { ... }
foo.bar = 3
```
While technically correct, it would still be useful if such variables were marked in some way, similar to how locals are reported as "never accessed" when they are only assigned to.

Of course, one tricky part here is that Lua may do metatable magic in the general case. So this check can only apply to tables that are the direct result of a `{ ... }` expression.

**Comment by mpeterv** [comment]
Good idea! Should be easy to implement by tracking regular accesses and field assignments separately.

**Comment by mpeterv** [comment]
Implemented on master branch:

```
Checking stdin                                    1 warning

    stdin:1:7: variable 'foo' is mutated but never accessed

Total: 1 warning / 0 errors in 1 file
```

**Comment by bjorn** [comment]
That's great! I'm looking forward to seeing where else our code is doing senseless computations. :-)
